mor/pyfeelpp-mor/feelpp/mor/pbdw.cpp

```diff
@@ -0,0 +1,62 @@
+//! Lesser General Public License for more details.
+//!
+//! You should have received a copy of the GNU Lesser General Public
+//! License along with this library; if not, write to the Free Software
+//! Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+//!
+//!
+#include <pybind11/pybind11.h>
+#include <pybind11/eigen.h>
+#include <pybind11/stl.h>
+
+#include <feel/feelmor/pbdw.hpp>
+
+namespace py = pybind11;
+
+PYBIND11_MODULE(_pbdw, m )
+{
+    using namespace Feel;
+    using pbdw_t = PBDWOnline;
+    using vectorN_type = typename pbdw_t::vectorN_type;
+    py::class_<pbdw_t,std::shared_ptr<pbdw_t>>(m, "Pbdw")
+        .def(py::init<std::string const&, int, std::string const&, std::string const&>(),
+             py::arg("name")=std::string("pbdw"),
+             py::arg("dbLoad")=1,
+             py::arg("dbFilename")=std::string(""),
+             py::arg("dbId")=std::string(""),
+             "Initialize PBDW"
+             )
+        .def("dimensionN", &pbdw_t::dimensionN, "Dimension of the Reduced Basis" )
+        .def("dimensionM", &pbdw_t::dimensionM, "Dimension of the sensors" )
+        .def("dimension", &pbdw_t::dimension, "Dimension of PBDW" )
+        .def("dimensionF", &pbdw_t::dimensionF, "Dimension of outputs" )
+        .def("matrix", &pbdw_t::matrix, "Matrix of PBDW" )
+        .def("matrixF", &pbdw_t::matrixF, "Matrix of outputs" )
+        .def("sensorNames", &pbdw_t::sensorNames, "Name of the sensors" )
+        .def("outputNames", &pbdw_t::outputNames, "Name of the outputs" )
+        .def("online", py::overload_cast<vectorN_type const&>(&pbdw_t::online, py::const_), "returns the coefficients of the solution", py::arg("yobs") )
+        .def("outputs", py::overload_cast<vectorN_type const&>(&pbdw_t::outputs, py::const_), "returns the outputs of the mode", py::arg("yobs") )
+        .def("online", py::overload_cast<vectorN_type const&, std::vector<int> const&, bool>(&pbdw_t::online, py::const_), "returns the coefficients of the solution", py::arg("yobs"), py::arg("sensors"), py::arg("toComplete")=true )
+        .def("outputs", py::overload_cast<vectorN_type const&, std::vector<int> const&, bool>(&pbdw_t::outputs, py::const_), "returns the outputs of the mode", py::arg("yobs"), py::arg("sensors"), py::arg("toComplete")=true )
+        .def("online", py::overload_cast<vectorN_type const&, std::vector<std::string> const&>(&pbdw_t::online, py::const_), "returns the coefficients of the solution", py::arg("yobs"), py::arg("sensors") )
+        .def("outputs", py::overload_cast<vectorN_type const&, std::vector<std::string> const&>(&pbdw_t::outputs, py::const_), "returns the outputs of the mode", py::arg("yobs"), py::arg("sensors") )
+        .def("onlineWithout", py::overload_cast<vectorN_type const&, std::vector<int> const&>(&pbdw_t::onlineWithout, py::const_), "returns the coefficients of the solution", py::arg("yobs"), py::arg("sensors") )
```

Comment:
Without what ?
the name is not very expressive

Reply:
also the documentation needs to be improved

Reply:
It is without the list of sensors given in argument. 
But I don't really use it in fact, maybe we can remove those methods and use only the one where we give the list of sensors used.